skrifa/src/outline/autohint/latin/edges.rs

```diff
@@ -39,77 +40,130 @@ pub(crate) fn compute_edges(
+                            break;
+                        }
+                        if let Some(next) = seg1.edge_next(&axis.segments) {
+                            seg1 = next;
+                        } else {
+                            break;
+                        }
+                    }
+                    if dist2 > edge_distance_threshold {
+                        continue;
+                    }
+                }
+                best_dist = dist;
+                best_edge_ix = Some(edge_ix);
             }
         }
-        // We couldn't find an edge, so add a new one for this segment
-        let opos = fixed_mul(segment.pos as i32, scale);
-        let edge = Edge {
-            fpos: segment.pos,
-            opos,
-            pos: opos,
-            dir: segment.dir,
-            first_ix: segment_ix as u16,
-            last_ix: segment_ix as u16,
-            ..Default::default()
-        };
-        axis.insert_edge(edge, top_to_bottom_hinting);
-        axis.segments[segment_ix].edge_next_ix = Some(segment_ix as u16);
-    }
-    // Loop again to find single point segments without a direction and
-    // associate them with an existing edge if possible
-    'segments2: for segment_ix in 0..axis.segments.len() {
-        let segment = &axis.segments[segment_ix];
-        if segment.dir != Direction::None {
-            continue;
+        if let Some(edge_ix) = best_edge_ix {
+            axis.append_segment_to_edge(segment_ix, edge_ix);
+        } else {
+            // We couldn't find an edge, so add a new one for this segment
+            let opos = fixed_mul(segment.pos as i32, scale);
+            let edge = Edge {
+                fpos: segment.pos,
+                opos,
+                pos: opos,
+                dir: segment.dir,
+                first_ix: segment_ix as u16,
+                last_ix: segment_ix as u16,
+                ..Default::default()
+            };
+            axis.insert_edge(edge, top_to_bottom_hinting);
+            axis.segments[segment_ix].edge_next_ix = Some(segment_ix as u16);
         }
-        // Find a matching edge
-        for edge_ix in 0..axis.edges.len() {
-            let edge = &axis.edges[edge_ix];
-            let dist = (segment.pos as i32 - edge.fpos as i32).abs();
-            if dist < edge_distance_threshold {
-                // We found an edge, link everything up
-                axis.append_segment_to_edge(segment_ix, edge_ix);
-                // Move to next segment
-                continue 'segments2;
+    }
+    if group == ScriptGroup::Default {
+        // Loop again to find single point segments without a direction and
+        // associate them with an existing edge if possible
+        'segments: for segment_ix in 0..axis.segments.len() {
+            let segment = &axis.segments[segment_ix];
+            if segment.dir != Direction::None {
+                continue;
+            }
+            // Find a matching edge
```

Comment:
Suggest `// Find the first matching edge`

Reply:
Changed to "Try to find an edge that coincides with this segment within the threshold" which better explains what's happening here